src/Service.Tests/Unittests/ConfigFileWatcherUnitTests.cs

```diff
@@ -159,5 +161,109 @@ public void HotReloadConfigRestRuntimeOptions()
+        {
+            // Arrange
+            int fileChangeNotificationsReceived = 0;
+            string configName = "ConfigFileWatcher_NotifiedOfOneNetNewChanges.json";
+            File.WriteAllText(configName, "MyValue");
+            ConfigFileWatcher fileWatcher = new(directoryName: Directory.GetCurrentDirectory(), configFileName: configName);
+            fileWatcher.NewFileContentsDetected += (sender, e) =>
+            {
+                // For testing, modification of fileChangeNotificationsRecieved
+                // should be atomic. 
+                Interlocked.Increment(ref fileChangeNotificationsReceived);
+            };
+
+            // Act - Write to the file twice to implicitly trigger
+            // multiple file change events for the ConfigFileWatcher's filesystem watcher.
+            ModifyConfigFile(configFileName: configName, configContent: "MyChangedValue");
+
+            // Assert -> allow time for the file change events to be processed.
+            // Wait time is arbitrary.
+            Thread.Sleep(millisecondsTimeout: 500);
+            Assert.AreEqual(expected: 1, actual: fileChangeNotificationsReceived);
+        }
+
+        /// <summary>
+        /// This test validates that overwriting a file with the same contents does not
+        /// result in ConfigFileWatcher's NewFileContentsDetected event being raised.
+        /// Internally, ConfigFileWatcher.NewFileContentsDetected is raised when the hash
+        /// of the file contents differ from what was previously detected.
+        /// In this test, the file contents are the same: MyValue -> MyValue.
+        /// The config file name is specific to this test in order to avoid concurrently
+        /// running tests from stepping over eachother due to acquiring a handle(s) to the file.
+        /// </summary>
+        [TestMethod]
+        public void ConfigFileWatcher_NotifiedOfZeroNetNewChange()
+        {
+            // Arrange
+            int fileChangeNotificationsReceived = 0;
+            string configName = "ConfigFileWatcher_NotifiedOfZeroNetNewChange.json";
+
+            // Write initial value to file.
+            File.WriteAllText(configName, "MyValue");
+
+            ConfigFileWatcher fileWatcher = new(directoryName: Directory.GetCurrentDirectory(), configFileName: configName);
+            fileWatcher.NewFileContentsDetected += (sender, e) =>
+            {
+                // For testing, modification of fileChangeNotificationsRecieved
+                // should be atomic. 
+                Interlocked.Increment(ref fileChangeNotificationsReceived);
+            };
+
+            // Act - Write to the file multiple times with the same value to implicitly trigger
+            // multiple file change events for the ConfigFileWatcher's filesystem watcher.
+            ModifyConfigFile(configFileName: configName, configContent: "MyValue");
+
+            // Assert -> allow time for the file change events to be processed.
```

Comment:
can we assert on more than 1 file change events detected instead of waiting for arbitrary amount of time? I wary of test getting flaky..

Reply:
Yep, good call. Added 2 invocations of the file-changed event to simulate how a real file system would notify our `ConfigFileWatcher` component. That way I'm directly demonstrating and asserting that `ConfigFileWatcher` only raises its `NewFileContentsDetected` event when a new file contents hash is determined, while validating that >1 filechange event was detected.